src/main/java/uk/gov/hmcts/reform/em/orchestrator/automatedbundling/AutomatedCaseUpdater.java

```diff
@@ -58,17 +59,29 @@ private void addNewBundle(BundleConfiguration configuration, ArrayNode bundles)
         bundle.setHasCoversheets(CcdBoolean.Yes);
         bundle.setHasTableOfContents(CcdBoolean.Yes);
         bundle.setFileName(configuration.filename);
-        int sortIndex = 0;
 
-        for (BundleConfigurationFolder folder : configuration.folders) {
+        addFolders(configuration.folders, bundle.getFolders(), 0);
+
+        bundles.add(bundleDtoToBundleJson(bundle));
+    }
+
+    private int addFolders(List<BundleConfigurationFolder> sourceFolders,
+                           List<CcdValue<CcdBundleFolderDTO>> destinationFolders,
+                           int sortIndex) {
+
+        for (BundleConfigurationFolder folder : sourceFolders) {
             CcdBundleFolderDTO ccdFolder = new CcdBundleFolderDTO();
             ccdFolder.setName(folder.name);
             ccdFolder.setSortIndex(sortIndex++);
 
-            bundle.getFolders().add(new CcdValue<>(ccdFolder));
+            destinationFolders.add(new CcdValue<>(ccdFolder));
```

Comment:
don't know whether this adding stuff to destinationFolders (which is a reference) is less readable than making bundle a class variable. I guess this is less stateful

Reply:
it's not that, it's because it's recursive and it adds the new folder to the parent folder rather than the top level bundle folders